app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.obligation
+import uk.gov.hmrc.gform.auth.models.MaterialisedRetrievals
+import uk.gov.hmrc.gform.gformbackend.GformConnector
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ FormTemplate, HmrcTaxPeriod }
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+case class HmrcTaxPeriodIdentifier(idType: String, idNumber: String, regimeType: String)
+
+class ObligationService(gformConnector: GformConnector) {
+
+  def lookupObligations(formTemplate: FormTemplate, retrievals: MaterialisedRetrievals)(
```

Comment:
On reflection, I think that this marshalling of obligations into a single object ought to be performed in the form backend